### PR TITLE
Set numpy and scipy as direct dependencies

### PR DESCRIPTION
**Pull request by victorlin**
### Description of proposed changes

These are currently installed as dependencies of dependencies. Since they are directly imported in this codebase, they should be direct dependencies with pinned versions.

### Testing

- [x] Checks pass

### Release tasks

- [ ] For the bioconda update, add numpy and scipy as dependencies in the [augur recipe](https://github.com/bioconda/bioconda-recipes/blob/master/recipes/augur/meta.yaml).

### Checklist

- [x] Add a message in [CHANGES.md](https://github.com/nextstrain/augur/blob/HEAD/CHANGES.md) summarizing the changes in this PR that are end user focused. Keep headers and formatting consistent with the rest of the file.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -57,10 +57,12 @@
         "isodate ==0.6.*",
         "jsonschema >=3.0.0, ==3.*",
         "networkx >= 2.5, ==2.*",
+        "numpy ==1.*",
         "packaging >=19.2",
         "pandas >=1.0.0, ==1.*",
         "phylo-treetime >=0.9.3, ==0.9.*",
         "pyfastx >=0.8.4, ==0.8.*",
+        "scipy ==1.7.*",
```
Comment (tsibley) [inline setup.py:65]: > Note: The latest version of scipy is 1.10.0, but 1.7.3 is the last version compatible with Python 3.7 which is still supported by Augur.

If we know Augur works with scipy 1.10.0, then we should be able to relax this restriction and let Pip's dependency resolver figure out a suitable scipy version depending on the Python version. e.g. maybe:

```suggestion
        "scipy ==1.*",
```

if we don't have any other lower bound.

**Review by tsibley — COMMENTED** [review]
If

File: setup.py (hunk 1 above)
Comment (victorlin) [reply setup.py:65]: Augur should work with scipy 1.10.0, but TreeTime doesn't and thus Augur CI fails with a new warning. I just took a deeper look and created this bug: https://github.com/neherlab/treetime/issues/223

My plan:

1. Wait for the warning to removed by a new TreeTime release which addresses the issue above.
2. Bump TreeTime's version in this PR.
3. Apply the suggestion above.
4. Re-open this PR for review.

**Comment by huddlej** [comment]
Good idea to add these dependencies explicitly. Part of the post-merge checklist is to update [the Bioconda recipe](https://github.com/bioconda/bioconda-recipes/blob/d300b51f6220dcaa424eb3e655af9e0fc62fe296/recipes/augur/meta.yaml#L23) accordingly.

I'd also vote for ignoring these warnings in the Augur functional test instead of waiting for Richard to make the fix in TreeTime and make a new release. The functional tests act a smoke tests (does `augur refine` run?) and also as a functional test (does the output exactly match what it should be?), but I don't expect these tests to catch changes in stderr or stdout unless that's explicitly a goal of a specific test (like when the "output" we want to match is the stdout or stderr text).

**Comment by victorlin** [comment]
@huddlej

I think it's beneficial to have tests that check for changes in stderr/stdout, to catch these kinds of external warnings (e.g. deprecation warnings) in CI. Although this one was from usage in TreeTime (and arguably should be caught by TreeTime CI), there's still a chance for warnings to surface from usage in Augur, which would prompt changes in Augur.

We could create separate tests where the goal is explicitly to check output, but I don't think it hurts to overload Cram tests like this for convenience of test authoring. It'd be tedious to maintain separate tests for (1) smoke testing, (2) functional testing, and (3) console output testing, especially when commands such as `augur refine` can be parameterized in so many different ways.

**Comment by victorlin** [comment]
I created and merged #1123, so this PR should be scoped to just adding direct dependencies. But since there is already discussion on the stderr issue, I'll summarize the plan here:

1. Temporarily ignore stderr output on affected steps (to prevent CI from being useless on other PRs).
2. Wait for a new release of TreeTime.
3. Bump the minimum version of TreeTime and revert the changes that ignored stderr.

I think the approach should work fine for other libraries as well. We'd just need to make sure this is followed through if/when new versions are released. Downside is that the new minimum version requirement might be too strict for some users – these can be evaluated on a case-by-case basis to determine if allowing older versions outweighs the benefits of capturing stderr of affected tests in CI.

File: setup.py (hunk 1 above)
Comment (victorlin) [reply setup.py]: Applied suggested change in a7413a3ed85c7fd8044fc43eefe5e9258b2bc058.

**Pull request merged**